pkg/request/batchstore.go

```diff
@@ -7,36 +7,34 @@ package request
 
 import (
 	"context"
-	"fmt"
 	"sync"
 	"sync/atomic"
-	"time"
 )
 
 type BatchStore struct {
-	batchingEnabled uint32
-	currentBatch    *batch
-	readyBatches    []*batch
-	onDelete        func(key string)
-	batchMaxSize    uint32
-	maxCapacity     int
-	keys2Batches    sync.Map
-	lock            sync.RWMutex
-	signal          sync.Cond
+	currentBatch      *batch
+	readyBatches      []*batch
+	onDelete          func(key string)
+	batchMaxSize      uint64
+	batchMaxSizeBytes uint64
+	keys2Batches      sync.Map
+	lock              sync.RWMutex
+	signal            sync.Cond
 }
 
 type batch struct {
-	lock     sync.RWMutex
-	size     uint32
-	enqueued bool
-	m        sync.Map
+	lock      sync.RWMutex
+	size      uint64
```

Comment:
1. why did we change it to uint64? Do you think 32 bits is not enough? 

2. Let's make both these uint64 limits - uint32, and then we can store them in a single uint64. This way, we can read them both atomically in a single operation (as well as increment them in a single operation - just add `2^32 +1`)

Reply:
I changed it to work with the configuration we have:
`RequestBatchMaxCount uint64` and `RequestBatchMaxBytes uint64`

Should I change the config as well?

Reply:
no. why change the config? Just cast from uint64 to uint32